### DISCUSSION TITLE
Polish the ctrl + z redo?

### DISCUSSION THREAD

**Discussion by JiaxinWang-thu** [microsoft/vscode-jupyter#9846] [New Ideas]
The present function of ctrl+z is that delete the current cell. However, it is dangerous since the wrong use of the ctrl+z will make the cell lost.

**Comment by amunger** [comment]
Is this the scenario you're talking about?
1. create a cell
2. write something in the cell
3. press escape to get out of edit mode
4. hit ctrl+z
Result: Cell with contents is removed

I agree that it is a bit shocking for that cell to suddenly be removed, though you should be able to hit redo (ctrl+y) to get that cell back.

It looks like https://github.com/microsoft/vscode/issues/139197 might be related (and probably should still be open)

**Reply by JiaxinWang-thu** [reply]
Yes. I suggest that you could remove such functionality from (ctrl+z), since there has already been 'delete the cell' button.
